mantidproject/mantid
# Improve visibility of sliceviewer and colorfill zoombox

**Description of work.**
This PR improves the visibility of the sliceviewer and colorfil zoombox. This is achieved by subclassing both the FigureCanvas and AxesImage. The FigureCanvas is subclassed so we can control the pen color and thickness used to draw the annotations from the toolbar. 

The AxesImage is subclassed to add a new method to find the intensity of the image in grayscale. Within this method the intensity varies between 0 and 255, where 0 indicates black (dark image) and 255 white (a light image). The value of grayscale yields the pen color (subject to a threshold), which will be used to draw the zoombox. This calculation is updated every time the image is drawn, e.g if we zoom on a high counting region the pen may change from white to black (depending on the color map).


**To test:**

-	Code review
-	Load MAR27284.nxs
-	Open sliceviewer
-	Test the magnification tool, it will be white to start with as the image is “dark”
-	Zoom in on the high counting data, as you go further in the tool will change to black (as yellow fills the screen)
-	Go back to the home view
-	Click reverse
-	The tool should now be black


Fixes #29264  






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
